pkg/kv/kvserver/rangefeed/scheduler.go

```diff
@@ -328,7 +328,7 @@ func (s *Scheduler) stopProcessor(id int64) {
 // Enqueue event for existing callback. The event is ignored if the processor
 // does not exist.
 func (s *Scheduler) enqueue(id int64, evt processorEventType) {
-	_, priority := s.priorityIDs.Load(id)
+	priority := s.priorityIDs.Contains(id)
```

Comment:
This is unsafe with `unregister`. There is no syncronization between the `priorityIDs` and the `shards`. I'm not sure how much of @wenyihu6 work will impact this, but maybe note a TODO that this has a synchronization risk...

There may be external constraints that make this safe, or its possible that we are OK just enquening on priority 0 if its not found, but this seems a little sketchy.